src/gui/wizard/xyzcalib.cpp

```diff
@@ -1,15 +1,145 @@
-// xyzcalib.c
-
-#include "xyzcalib.h"
+// xyzcalib.cpp
+#include "i18n.h"
 #include "gui.hpp"
-#include "dbg.h"
-#include "config.h"
-#include "stm32f4xx_hal.h"
-#include "marlin_client.h"
-#include "wizard_config.h"
-#include "screen_wizard.h"
-#include "wizard_ui.h"
-#include "window_dlg_calib_z.hpp"
+#include "xyzcalib.hpp"
+
+StateFncData StateFnc_XYZCALIB_INIT(StateFncData last_run) {
+    static const char *title_txt = N_(
+        "State              \n"
+        "XYZCALIB_INIT      \n"
+        "not implemented");
+    static const string_view_utf8 title = string_view_utf8::MakeCPUFLASH((const uint8_t *)(title_txt));
+
+    MsgBox(title, Responses_NEXT);
+    return last_run.PassToNext();
+}
+
+StateFncData StateFnc_XYZCALIB_HOME(StateFncData last_run) {
+    static const char *title_txt = N_(
+        "State              \n"
+        "XYZCALIB_HOME      \n"
+        "not implemented");
+    static const string_view_utf8 title = string_view_utf8::MakeCPUFLASH((const uint8_t *)(title_txt));
+
+    MsgBox(title, Responses_NEXT);
+    return last_run.PassToNext();
+}
+
+StateFncData StateFnc_XYZCALIB_Z(StateFncData last_run) {
+    static const char *title_txt = N_(
+        "State              \n"
+        "XYZCALIB_Z         \n"
+        "not implemented");
+    static const string_view_utf8 title = string_view_utf8::MakeCPUFLASH((const uint8_t *)(title_txt));
+
+    MsgBox(title, Responses_NEXT);
+    return last_run.PassToNext();
+}
+
+StateFncData StateFnc_XYZCALIB_XY_MSG_CLEAN_NOZZLE(StateFncData last_run) {
+    static const string_view_utf8 title = _(
+        "Please clean the nozzle "
+        "for calibration. Click "
+        "NEXT when done.");
+
+    MsgBox(title, Responses_NEXT);
+    return last_run.PassToNext();
+}
+
+StateFncData StateFnc_XYZCALIB_XY_MSG_IS_SHEET(StateFncData last_run) {
+    static const string_view_utf8 title = _(
+        "Is steel sheet "
+        "on heatbed?");
+
+    if (MsgBox(title, Responses_YesNo) == Response::Yes) {
+        return StateFncData(WizardState_t::XYZCALIB_XY_MSG_REMOVE_SHEET, WizardTestState_t::PASSED);
+    } else {
+        return StateFncData(WizardState_t::XYZCALIB_XY_MSG_PLACE_PAPER, WizardTestState_t::PASSED);
+    }
+    return last_run.PassToNext();
+}
+
+StateFncData StateFnc_XYZCALIB_XY_MSG_REMOVE_SHEET(StateFncData last_run) {
+    static const string_view_utf8 title = _(
+        "Please remove steel "
+        "sheet from heatbed.");
+
+    MsgBox(title, Responses_NEXT);
+    return last_run.PassToNext();
+}
+
+StateFncData StateFnc_XYZCALIB_XY_MSG_PLACE_PAPER(StateFncData last_run) {
+    static const string_view_utf8 title = _(
```

Comment:
this cannot be static const, the current language may change during the runtime of the machine